plugins/modules/ip_rdns.py

```diff
@@ -0,0 +1,126 @@
+from ansible_collections.linode.cloud.plugins.module_utils.linode_common import (
+    LinodeModuleBase,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_docs import (
+    global_authors,
+    global_requirements,
+)
+from ansible_specdoc.objects import FieldType, SpecDocMeta, SpecField, SpecReturnValue
+from linode_api4 import IPAddress
+
+ip_rdns_spec = dict(
+    # Disable the default values
+    label=SpecField(type=FieldType.string, required=False, doc_hide=True),
+
+    state=SpecField(
+        type=FieldType.string,
+        choices=["present", "absent"],
+        description=["The state of this rDNS of the IP address."],
+    ),
+    address=SpecField(
+        type=FieldType.string,
+        description=["The IP address."],
+        required=True,
+    ),
+    rdns=SpecField(
+        type=FieldType.string,
+        editable=True,
+        description=["The desired rDNS value."],
+    ),
+)
+
+SPECDOC_META = SpecDocMeta(
+    description=["Manage a Linode IP address's rDNS."],
+    requirements=global_requirements,
+    author=global_authors,
+    options=ip_rdns_spec,
+    examples=ip_rdns_docs.specdoc_examples,
+    return_values=dict(
+        ip=SpecReturnValue(
+            description=(
+                "The updated IP address with the new "
+                "reverse DNS in JSON serialized form."
+            ),
+            docs_url=(
+                "https://www.linode.com/docs/api/profile/#ip-address-rdns-update"
+            ),
+            type=FieldType.dict,
+            sample=ip_docs.result_ip_samples,
+        )
+    ),
+)
+
+
+class ReverseDNSModule(LinodeModuleBase):
+    """Module for updating Linode IP address's reverse DNS value"""
+
+    def __init__(self) -> None:
+        self.module_arg_spec = SPECDOC_META.ansible_spec
+        self.required_one_of = ["state", "rdns"]
+        self.results = dict(
+            changed=False,
+            actions=[],
+            ip=None,
+        )
+        super().__init__(
+            module_arg_spec=self.module_arg_spec,
+            required_if=[["state", "present", ["rdns"]]],
+        )
+
+    def update_rdns(self, rdns: str) -> None:
+        """
+        Update the reverse DNS of the IP address.
+        """
+        ip_str = self.module.params.get("address")
+        ip_obj = self._get_resource_by_id(IPAddress, ip_str)
+        ip_obj.rdns = rdns
+        try:
+            ip_obj.save()
+            ip_obj._api_get()
+            self.register_action(
+                f"Updated reverse DNS of IP address {ip_str} to be {rdns}"
+            )
+        except Exception as exception:
+            self.fail(f"Failed to update reverse DNS of IP address {ip_str}, {exception}")
+        self.results["ip"] = ip_obj._raw_json
+
+    def _handle_present(self) -> None:
+        rdns = self.module.params.get("rdns")
+        if not rdns:
+            self.fail("`rdns` attribute is required to update the IP address")
+        else:
```

Comment:
Tiny nitpick but I think you can remove the `else` here :+1:

Reply:
Good catch! Thank you.